apiv6/kentikapi/httputil/httputil.go

```diff
@@ -93,22 +90,21 @@ func defaultHTTPClient() *http.Client {
 func makeRetryPolicy(statusCodes []int, methods []string) retryablehttp.CheckRetry {
 	statusCodesSet := makeIntSet(statusCodes)
 	methodsSet := makeStringSet(methods)
-	redirectsErrorRe := redirectsErrorRegexp()
-	schemeErrorRe := schemeErrorRegexp()
 
 	return func(ctx context.Context, resp *http.Response, err error) (bool, error) {
-		// do not retry on context.Canceled or context.DeadlineExceeded
 		if ctx.Err() != nil {
 			return false, ctx.Err()
 		}
 
-		if !isRequestRetrayable(resp.Request, methodsSet) {
-			return false, err
+		if err != nil {
+			if isErrorTemporary(err) {
+				return true, nil
+			}
+			return false, nil
 		}
 
-		if err != nil {
-			// TODO(dfurman): unit test that branch
-			return isErrorRecoverable(err, redirectsErrorRe, schemeErrorRe)
+		if !isRequestRetrayable(resp.Request, methodsSet) {
```

Comment:
typo: "IsRequestRetrayable" should be "IsRequestRetryable"

Reply:
Will be fixed in next PR